Looped/Protocols/FlowController.swift

```diff
@@ -0,0 +1,5 @@
+import UIKit
+
+protocol FlowController {
+    var rootViewController: UINavigationController { get }
```

Comment:
Najprawdopodobniej bedzie tak, ze nie zawsze ekran bedzie mial navigation controller wtedy Twoj protokol jest za malo elastycxzny. Co powiesz na:

```swift


protocol FlowControler {
    associatedtype RootViewController

    var rootViewController: RootViewController { get }
}

class A: FlowControler {
    typealias RootViewController = UINavigationController

    var rootViewController: UINavigationController
}
```

Reply:
To jest świetne rozwiązanie, którego mi brakowało. Protokół FlowController wymusza deklarowanie rootViewController typu RootViewController, który może będzie dopiero określony! Zgadza się? Bomba.

Reply:
Dokladnie tak jest jak mowisz :)